src/main/java/uk/gov/hmcts/cmc/claimstore/services/staff/ClaimIssuedStaffNotificationService.java

```diff
@@ -28,45 +26,38 @@
     private final EmailService emailService;
     private final StaffEmailProperties staffEmailProperties;
     private final ClaimIssuedStaffNotificationEmailContentProvider provider;
-    private final DefendantPinLetterPdfService defendantPinLetterPdfService;
-    private final SealedClaimDocumentService sealedClaimDocumentService;
 
     @Autowired
     public ClaimIssuedStaffNotificationService(
         final EmailService emailService,
         final StaffEmailProperties staffEmailProperties,
-        final ClaimIssuedStaffNotificationEmailContentProvider provider,
-        final DefendantPinLetterPdfService defendantPinLetterPdfService,
-        final SealedClaimDocumentService sealedClaimDocumentService
+        final ClaimIssuedStaffNotificationEmailContentProvider provider
     ) {
         this.emailService = emailService;
         this.staffEmailProperties = staffEmailProperties;
         this.provider = provider;
-        this.defendantPinLetterPdfService = defendantPinLetterPdfService;
-        this.sealedClaimDocumentService = sealedClaimDocumentService;
     }
 
-    public void notifyStaffClaimIssued(
-        final Claim claim,
-        final String defendantPin,
-        final String submitterEmail
-    ) {
-        requireNonNull(claim);
-        requireNonBlank(submitterEmail);
-        final EmailData emailData = prepareEmailData(claim, defendantPin, submitterEmail);
+    @EventListener
+    public void notifyStaffOfClaimIssue(final DocumentGeneratedEvent event) {
+        requireNonNull(event);
+
+        final EmailData emailData = prepareEmailData(event.getClaim(), event.getDocuments());
         emailService.sendEmail(staffEmailProperties.getSender(), emailData);
     }
 
     private EmailData prepareEmailData(
         final Claim claim,
-        final String defendantPin,
-        final String submitterEmail
-    ) {
-        EmailContent emailContent = provider.createContent(wrapInMap(claim));
+        final List<PDF> documents) {
```

Comment:
I'm a little worried about having PDF everywhere instead of a nicer abstraction of a file/document. Might make refactoring a pain if a different file type is introduced later.

Reply:
We can think about `PDF` being more concrete implementation of `Document` class. I could create interface but there would be just single implementation. I think it would not be much difficult to do it later especially that we are not building public API here.

Reply:
Ok, fair enough 👍